File: bindings/java-jni/src/main/java/org/maplibre/nativejni/style/StyleImage.java
package org.maplibre.nativejni.style;

import java.util.Objects;
import org.maplibre.nativejni.render.PremultipliedRgba8Image;

/** Copied runtime style image pixels with style image metadata. */
public record StyleImage(PremultipliedRgba8Image image, float pixelRatio, boolean sdf) {
  public StyleImage {
    Objects.requireNonNull(image, "image");
  }
}

File: bindings/kotlin/src/commonMain/kotlin/org/maplibre/nativeffi/style/StyleImage.kt
package org.maplibre.nativeffi.style

import org.maplibre.nativeffi.render.PremultipliedRgba8Image

/** Copied runtime style image pixels with style image metadata. */
public data class StyleImage(
  public val image: PremultipliedRgba8Image,
  public val pixelRatio: Float,
  public val sdf: Boolean,
)
